gemrb/core/Map.h

```diff
@@ -196,8 +196,13 @@ class MapNote {
 		if (text) {
 			//update custom strref
 			char* mbstring = MBCStringFromString(*text);
-			strref = core->UpdateString( strref, mbstring);
-			free(mbstring);
+			if (mbstring) {
+				strref = core->UpdateString( strref, mbstring);
+				free(mbstring);
+			} else {
+				strref = core->UpdateString(strref, "?");
+				Log(WARNING, "Map", "Failed to update string from map note, possibly an enconding issue.");
```

Comment:
I think it will be good log also `strref`, because this log only indicates problem _somewhere_, but is useless when you want fix this problem

Reply:
This can only happen with custom map notes, so it will all be in the TLK override. With a brand new strref, so I'm not sure it'd be that useful.

Reply:
It wasn't even a custom map note, but one from the string table containing a `ß` (ISO 8859-1 `0xDF`). At some point in `String.cpp`, it is converted into a wide char string by this https://github.com/gemrb/gemrb/blob/89b2bfa1cf8ec8dd3b543bfbb9c79fbd6c96c60b/gemrb/core/System/String.cpp#L87 Trying to get this through `wcstombs` fails on my Linux.

Reply:
For "why", I think you nailed it. Unless the original disabled it, we shouldn't patch over it.

As for the underlying issue, it still reminds me of the Czech and Polish versions, which also used a different encoding (not sure if even standard), which we then remap. See here: https://github.com/gemrb/gemrb/blob/master/gemrb/docs/en/default_ini.txt